src/Yaml/Parser.elm

```diff
@@ -312,24 +314,13 @@ recordOrString indent indent_ =
                 ]
 
         addRemaining string remaining =
-            Ast.fromString <| U.postProcessString (removeComment string ++ remaining)
-
-        removeComment string =
-            string
-                |> String.split "#"
-                |> List.head
-                |> Maybe.withDefault ""
+            Ast.String_ (string ++ remaining)
     in
     P.oneOf
-        [ quotedString indent_
-        , P.succeed identity
-            |. P.chompIf (U.neither U.isColon U.isNewLine)
-            |. P.chompWhile (U.neither U.isColon U.isNewLine)
-            |> P.getChompedString
-            |> P.andThen withString
+        [ P.succeed Ast.Null_
+            |. P.end
         , P.succeed identity
-            |. P.chompWhile U.isColon
-            |> P.getChompedString
+            |= U.characters (not << U.isColon)
```

Comment:
I think I see what you're going for here but `Yaml.Parser.Util` doesn't export the `characters` function. Consider exporting that function for use here